operator/src/main/java/com/instaclustr/cassandra/operator/service/ControllerService.java

```diff
@@ -121,25 +89,36 @@ void statefulSetEvent(final StatefulSetEvent event) {
     @Override
     protected void run() throws Exception {
         while (isRunning()) {
-            final DataCenterKey dataCenterKey = dataCenterQueue.take();
-            
-            if (dataCenterKey == POISON)
-                return;
-            
-            final DataCenter dataCenter = dataCenterCache.getIfPresent(dataCenterKey);
-
-            if (dataCenter == null) {
-                appsApi.deleteNamespacedStatefulSet(dataCenterKey.name, namespace, null, null, null, null, null);
-
-                continue;
-            }
+        	processNextDataCenterEvent();
+        	//processNextClusterEvent();
+        	//processNextSecretEvent();
+        	//processNextStatefulSetEvent();
+        }
+    }
+    
+    private void processNextDataCenterEvent() throws Exception {
+    	final DataCenterKey dataCenterKey = dataCenterQueue.take();
+        
+        if (dataCenterKey == POISON) return;
```

Comment:
This check needs to be in the main loop, rather than in a separate function. The intent of the `POISON` marker is to abort the loop.

Reply:
add a comment here only for the format issue. Looks like it's related with my IDE. Reformatting the whole file will mess up with statefulset specs and other methods. might need some helps on this.